python/res/fm/ecl/ecl_run.py

```diff
@@ -164,77 +172,73 @@ def initMPI(self):
                 fileH.write("%s\n" % host)
 
 
-
-    def initRedirect(self):
+    def execSerialEclipse(self):
         stdout_file = "%s.LOG" % self.base_name
         stderr_file = "%s.stderr" % self.base_name
-
-        # Redirecting stdout / stderr
-        fd_stdout = os.open(stdout_file , os.O_WRONLY | os.O_TRUNC | os.O_CREAT , 0o644);
-        fd_stderr = os.open(stderr_file , os.O_WRONLY | os.O_TRUNC | os.O_CREAT , 0o644);
-
-        os.dup2(fd_stdout , 1)
-        os.dup2(fd_stderr , 2)
-
-        os.close(fd_stdout)
-        os.close(fd_stderr)
-
-
-
-    def execSerialEclipse(self):
-        os.execv( self.sim.executable , [ self.sim.executable , self.base_name ])
+        with open(stdout_file, "w") as stdout, open(stderr_file, "w") as stderr:
+            process = subprocess.Popen(
+                [self.sim.executable, self.base_name],
+                env=self.sim.env,
+                stdout=stdout,
+                stderr=stderr
+            )
+            process.communicate()
+            return process.returncode
 
 
     def execParallellEclipse(self):
         self.initMPI( )
-        os.execv( self.sim.mpirun , [self.sim.mpirun , "-machinefile" , self.machine_file , "-np" , "%s" % self.num_cpu , self.sim.executable , self.base_name ] )
+        stdout_file = "%s.LOG" % self.base_name
+        stderr_file = "%s.stderr" % self.base_name
+        with open(stdout_file, "w") as stdout, open(stderr_file, "w") as stderr:
+            process = subprocess.Popen(
+                [self.sim.mpirun,
+                 "-machinefile",
+                 self.machine_file,
+                 "-np",
+                 "%s" % self.num_cpu,
+                 self.sim.executable,
+                 self.base_name],
+                env=self.sim.env,
+                stdout=stdout,
+                stderr=stderr)
```

Comment:
This is getting really picky; so besides making the line breaking of the `Popen` calls on the same format (one is `stderr)` and the other is `stderr\n)`) feel free to ignore my comments below.

Personally I like
```python
process = subprocess.Popen(
    ......,
    stdout=stdout,
    stderr=stderr,
)
```
because there is nothing special about the last arguments. I find it easier to read. In addition it is more friendly with respect to git history; because you can inject another argument and get a single plus line.

In addition, I'm not a big fan of non-standard indents used for code alignment, as they quickly go out of sync. My personal preference, as well as _black_'s preference (the tool we use on the repositories we have activated automatic code formatting) is the following if you are to split a list over multiple lines:
```python
process = subprocess.Popen(
    [
        self.sim.mpirun,
        "-machinefile",
        self.machine_file,
        "-np",
        "%s" % self.num_cpu,
        self.sim.executable,
        self.base_name,
    ],
    env=self.sim.env,
    stdout=stdout,
    stderr=stderr,
)
```
As stated above, do with these comments as you like ;)

Reply:
Completely agree, I just wish we could start using black so that we could enforce this everywhere. It's easy to forget the formatting when I can't just format the entire file.

Reply:
I agree. This one would be a start: https://github.com/equinor/libres/issues/701 ;)